packages/integrations/vercel/src/serverless/adapter.ts

```diff
@@ -207,14 +207,31 @@ You can set functionPerRoute: false to prevent surpassing the limit.`
 
 				// Multiple entrypoint support
 				if (_entryPoints.size) {
-					for (const [route, entryFile] of _entryPoints) {
-						const func = basename(entryFile.toString()).replace(/\.mjs$/, '');
+					// TODO: Currently this function directly checks for a single route 
+					// injected in packages/astro/src/assets/internal.ts Ideally this
+					// should be done dynamically
+					const isInjectedRoute = (route: RouteData) =>
+						route.pathname === '/_image';
+
+          const getInjectedFuncName = (entryFile: URL) =>
+            basename(entryFile.toString())
+              .replace(/^entry\./, '')
+              .replace(/\.mjs$/, '');
```

Comment:
Why a regex? Isn't `replace("entry.", "")` enough?

Reply:
This regex only matches "`entry.`" at the beginning of the string. You could ask the same question about the original `.replace(/\.mjs$/, '');` code. It could use `.replace(".mjs", "")` too.

Reply:
Regarding your other comment, if we rename `isInjectedRoute` to `isImageRoute`, it would be natural to rename `getInjectedFuncName` to `getImageFuncName` too, then this function could return a fixed string.

Reply:
Sorry, I ask the question in a different way: Why did you use a regex instead of a simple string? Do they have the same impact performance-wise?

Reply:
I used it to ensure precision. For example, it avoids incorrectly replacing 'hello.world.entry.astro.mjs' when `entryFile` doesn't start with 'entry.'. Regex allowed me to target replacements specifically at string's start. However, if you are certain that it always begins with `entry.` then that's enough!

Simple string replacement is generally faster, though the speed difference in this case is most likely negligible.

I am not that attached to this regex, I removed it. I just want this PR to get merged so I can upgrade to v3.